packages/aws-cdk-lib/aws-apigatewayv2-integrations/lib/http/sqs.ts

```diff
@@ -0,0 +1,117 @@
+import * as apigwv2 from '../../../aws-apigatewayv2';
+import * as iam from '../../../aws-iam';
+import * as sqs from '../../../aws-sqs';
+
+/**
+ * Properties to initialize `HttpSqsIntegration`.
+ */
+export interface HttpSqsIntegrationProps {
+  /**
+   * Specifies how to transform HTTP requests before sending them to the backend.
+   *
+   * When the subtype is either `START_EXECUTION` or `START_SYNC_EXECUTION`,
+   * it is necessary to specify the `StateMachineArn`.
+   * Conversely, when the subtype is `STOP_EXECUTION`, the `ExecutionArn` must be specified.
+   *
+   * @see https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-parameter-mapping.html
+   *
+   * @default - specify only `StateMachineArn`
+   */
+  readonly parameterMapping?: apigwv2.ParameterMapping;
+
+  /**
+   * The subtype of the HTTP integration.
+   *
+   * Only subtypes starting with SQS_ can be specified.
+   *
+   * @default HttpIntegrationSubtype.SQS_SEND_MESSAGE
+   */
+  readonly subtype?: apigwv2.HttpIntegrationSubtype;
```

Comment:
Since only a couple of these are relevant here, I think that this isn't quite the right type to use here. Of the 10 values, only 4 work.

Additionally, per the comments above where you mention that certain arns need to be added based off the subtype here, I think that these props aren't quite the right contract in general. For instance, in the case that this subtype is `START_EXECUTION`, the parameter mapping shouldn't be optional because it must include at least the `StateMachineArn`.

Reply:
Wait, I actually went back and read through the SubTypes more and it looks like the subtypes listed in the `parameterMapping` docstring above don't start with `SQS_` so those comments don't mesh with the ones here in `subType`.

Reply:
I'm really sorry for describing crazy comments for `parameterMapping`. 
I have updated it and could you please re-confirm it?

Reply:
Why not narrow down the type to the sane choices?

```ts
readonly subtype: Pick<
  apigwv2.HttpIntegrationSubtype,
  'SQS_SEND_MESSAGE' | 'SQS_RECEIVE_MESSAGE' | 'SQS_DELETE_MESSAGE' | 'SQS_PURGE_QUEUE'
>
```

Reply:
@chriswilty Thank you for your suggestion.
This `PickEnum` type will work correctly in TypeScript. However, it may not function properly when converted to other languages using jsii. 
For example, I believe even TypeScript's union types couldn't be used in CDK.

I would like to check with the maintainer if it is usable.